test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -4,183 +4,223 @@
 
-    # Set up call returns. These calls are in the order they occur in the code
-    sdkmr.check_as_admin.return_value = True
-    sdkmr.save_job.return_value = job_id
-    ws.get_object_info3.return_value = {"paths": [[ws_obj1], [ws_obj2]]}
-    catalog_resources = {
-        "client_group": client_group,
-        "request_cpus": cpus,
-        "request_memory": mem,
+    return mocks
+
+
+def _set_up_common_return_values(mocks):
+    """
+    Set up return values on mocks that are the same for several tests.
+    """
+    mocks[Workspace].get_object_info3.return_value = {
+        "paths": [[_WS_REF_1], [_WS_REF_2]]
     }
-    catutils.get_normalized_resources.return_value = catalog_resources
-    condor.extract_resources.return_value = CondorResources(
-        cpus, "2600GB", mem, client_group
-    )
-    catalog.get_module_version.return_value = {"git_commit_hash": git_commit}
-    condor.run_job.return_value = SubmissionInfo(cluster, {}, None)
+    mocks[SDKMethodRunner].save_job.return_value = _JOB_ID
+    mocks[Catalog].get_module_version.return_value = {"git_commit_hash": _GIT_COMMIT}
+    mocks[Condor].run_job.return_value = SubmissionInfo(_CLUSTER, {}, None)
     retjob = Job()
-    retjob.id = ObjectId(job_id)
-    retjob.status = created_state
-    mongo.get_job.return_value = retjob
+    retjob.id = ObjectId(_JOB_ID)
+    retjob.status = _CREATED_STATE
+    mocks[MongoUtil].get_job.return_value = retjob
 
-    # set up the class to be tested and run the method
-    rj = EE2RunJob(sdkmr)
-    params = {
-        "method": method,
-        "source_ws_objects": [ws_obj1, ws_obj2],
-    }
-    assert rj.run(params, as_admin=True) == job_id
 
-    # check mocks called as expected. The order here is the order that they're called in the code.
-    sdkmr.check_as_admin.assert_called_once_with(JobPermissions.WRITE)
-    ws.get_object_info3.assert_called_once_with(
-        {"objects": [{"ref": ws_obj1}, {"ref": ws_obj2}], "ignoreErrors": 1}
+def _check_common_mock_calls(mocks, reqs, creqs, wsid):
+    """
+    Check that mocks are called as expected when those calls are similar or the same for
+    several tests.
+    """
+    sdkmr = mocks[SDKMethodRunner]
+    kafka = mocks[KafkaClient]
+    mocks[Workspace].get_object_info3.assert_called_once_with(
+        {"objects": [{"ref": _WS_REF_1}, {"ref": _WS_REF_2}], "ignoreErrors": 1}
     )
-    catutils.get_normalized_resources.assert_called_once_with(method)
-    condor.extract_resources.assert_called_once_with(catalog_resources)
-    catalog.get_module_version.assert_called_once_with(
+    mocks[Catalog].get_module_version.assert_called_once_with(
         {"module_name": "lolcats", "version": "release"}
     )
 
     # initial job data save
     expected_job = Job()
-    expected_job.user = user
-    expected_job.status = created_state
+    expected_job.user = _USER
+    expected_job.status = _CREATED_STATE
+    expected_job.wsid = wsid
```

Comment:
worth making into a variable? it's used a couple of lines further down.

Reply:
Is this comment on the wrong line maybe? `wsid` is a variable...

Reply:
Resolved this offline, leaving as is for now, anyway